web/frontend/src/components/gantry/index.svelte

```diff
@@ -50,6 +64,50 @@ const increment = (axis: number, amount: number) => {
   $robotClient.gantryService.moveToPosition(req, displayError);
 };
 
+const gantryModifyAllDoMoveToPosition = () => {
+  const gantry = status!;
+  const newList : number[] = Array.from({ length: gantry.positions_mm.values.length });
+  const newPieces = modifyAllStatus.pieces;
+
+  for (const [i, newPiece] of newPieces.entries()) {
+    newList[i] = newPiece!.pos;
+  }
+
+  const req = new gantryApi.MoveToPositionRequest();
+  req.setName(name);
+  req.setPositionsMmList(newList);
+
+  rcLogConditionally(req);
+  $robotClient.gantryService.moveToPosition(req, displayError);
+
+  modifyAll = false;
+};
+
+const gantryHome = () => {
+  const req = new gantryApi.HomeRequest();
+  req.setName(name);
+
+  rcLogConditionally(req);
+  $robotClient.gantryService.home(req, displayError);
+};
+
+const gantryModifyAll = () => {
+  const nextPiece = [];
+
+  for (const part of parts) {
+    nextPiece.push({
+      axis: part!.axis,
```

Comment:
It doesn't seem like there's any case where part can be undefined, is it possible to remove the `!` on these lines?